era_5g_interface/channels.py

```diff
@@ -234,14 +238,18 @@ def send_data(
             new_data = compress(bytes(ujson.dumps(data), "utf-8"))
 
         if isinstance(self._sio, socketio.Client):
-            if not self._sio.connected:  # self.eio.state == 'connected'
-                raise ConnectionError("Client is not connected to server.")
+            if wait_for_reconnection:
+                while not self._sio.connected or not self._sio.eio.state:
```

Comment:
Is it ok to wait forever?

Reply:
I can change it to:
```
                    if not self._sio._reconnect_task or not self._sio._reconnect_task.is_alive():
                        break
```
It will not wait forever, it depends on _reconnect_task

Reply:
I'm not sure how it works, it's up to you :-)